torch/_logging/structured.py

```diff
@@ -22,6 +25,23 @@ def intern_string(s: str) -> int:
     return r
 
 
+def dump_file(filename: str) -> None:
```

Comment:
instead of rolling out your own check, why not use functools.lru_cache?

Reply:
I didn't think about lru_cache when I wrote the code, but in this case, there's a substantive difference, because I don't want to dump file names that aren't eval_with_key into the cache. Also... idk, it's three LOC, is a decorator really better lol.